### DISCUSSION TITLE
Mobile App

### DISCUSSION THREAD

**Discussion by XenoUniv3rse** [open-webui/open-webui#3031] [General]
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
I know that I can just add the webpage to my home screen on iOS and will act as "Native" but I would like to have an actual application that I can download from the app store. After download, allow users to enter the address to their Open WebUI service. I think that will give users more of a "native" feel. 

**Additional context**
Thanks for all the hard work!! What an amazing project!

**Comment by justinh-rahb** [comment]
App Store approval is a right pain in the rear. We're not very interested in experiencing it right now.

Related:
- #470

**Reply by XenoUniv3rse** [reply]
That sucks, but I completely get that. Getting apps on the APP store is a real pain in the ass. I was looking at Enchanted (https://github.com/AugustDev/enchanted) and its real nice, but not a as convenient, or has as many features as Open WebUI. An Open WebUI + Enchanted baby would be amazing! 

Keep up the great work!

**Reply by XenoUniv3rse** [reply]
Just one more reason why an app would be great. 

There are a few things that I would love to do with it in the Apple Shortcuts app. 

Just a mention

**Reply by Anaphylaxis** [reply]
Any name ideas for anyone that makes an openai-api-compatible iOS client? That they don't want the rights to 🤪

**Reply by XenoUniv3rse** [reply]
No name idea's... but... are you working on something?😁

**Reply by haydonryan** [reply]
It sounds like it's a feature request   / PR for enchanted to  support openwebui

**Reply by haydonryan** [reply]
I created a Feature Request on enchanted's github project: https://github.com/gluonfield/enchanted/issues/208.

I was surprised when I searched that I couldn't find anything, but maybe my search fu is turned off atm.

**Reply by Anaphylaxis** [reply]
> I created a Feature Request on enchanted's github project: [gluonfield/enchanted#208](https://github.com/gluonfield/enchanted/issues/208).
> 
> I was surprised when I searched that I couldn't find anything, but maybe my search fu is turned off atm.

instead of supporting "open-webui", it should support "openai api", which owui and a lot of other things use. if you searched openai you'd find plenty already.

**Reply by haydonryan** [reply]
Doh, thanks.

**Comment by thearyadev** [comment]
i wonder if targeting the side-load side would be suitable. 

making an app for the app store is a ton of work, but creating one that could be installed using something like AltStore could still provide an option for those who would benefit a lot from a mobile app, without the apple process

but then again, thats likely a very niche selection of people.

**Comment by GylanSalih** [comment]
to create a native iOS app for thee Open WebUI service, here’s a streamlined approach:

   ```
 Build the App: Use Swift or Objective-C to develop the app. It will let users enter and save their Open WebUI service address.
    Integrate WebView: Use WKWebView to display the Open WebUI seervice in the app, giving it a native feel.
    Save Addresses: Implement a feature to save and manage multiple service addresses, with options for local storage or iCloud syncing.
    Offline Features: Consider adding offline capabilities or caching for a smoother experience.
    App Store Ready: Make sure the app follows Apple’s guidelines for privacy and security.
    Test and Refine: Thoroughly test the app and get user feedback to make improvements before launch.
```

**Comment by cyberj0g** [comment]
I believe the app will be needed to support extended voice interactions with open-source multi-modal LLMs, which are rapidly becoming available. To be truly useful, such assistants must constantly listen in the background, without requiring a wake word. Which is not possible with the web app, but certainly coming soon to Apple Intelligence, Google Assistant etc. It's crucial to have an open alternative, for those of us unwilling to hand over control of such sensitive data to corporations. It may even be worth considering a platform switch if the app would be available only for Android, so I would suggest adding the app to the roadmap.

**Reply by Xyz00777** [reply]
If you don't want to have the hustle with the official play store than why not fdroid at least for Android

**Comment by neuromechanist** [comment]
I don't know of you have seen [Pal](https://apps.apple.com/app/id6447545085), it is quite impressive, with just a solo developer.

I am just an enthusiastic user, but it would be cool to evlevat Pal with some Openwebui UX and backend.

**Reply by spammenotinoz** [reply]
> I don't know of you have seen [Pal](https://apps.apple.com/app/id6447545085), it is quite impressive, with just a solo developer.
> 
> I am just an enthusiastic user, but it would be cool to evlevat Pal with some Openwebui UX and backend.

Yes, unsure why you received a thumbs-down, this is a decent iOS Application which I use with Open-WebUI.
Only downside I can think of, is for most end-users, it's a bit difficult to configure.

**Reply by mariomaz87** [reply]
> > I don't know of you have seen [Pal](https://apps.apple.com/app/id6447545085), it is quite impressive, with just a solo developer.
> > I am just an enthusiastic user, but it would be cool to evlevat Pal with some Openwebui UX and backend.
> 
> Yes, unsure why you received a thumbs-down, this is a decent iOS Application which I use with Open-WebUI. Only downside I can think of, is for most end-users, it's a bit difficult to configure.

How do you use Pal with OpenWeb Ui?

**Reply by spammenotinoz** [reply]
> > > I don't know of you have seen [Pal](https://apps.apple.com/app/id6447545085), it is quite impressive, with just a solo developer.
> > > I am just an enthusiastic user, but it would be cool to evlevat Pal with some Openwebui UX and backend.
> > 
> > 
> > Yes, unsure why you received a thumbs-down, this is a decent iOS Application which I use with Open-WebUI. Only downside I can think of, is for most end-users, it's a bit difficult to configure.
> 
> How do you use Pal with OpenWeb Ui?

Scroll down to advanced, then you see the option to "Provide custom model\host\endpoint" then use /api instead of the usual /v1

**Reply by mlazzarotto** [reply]
@spammenotinoz Did you have to provide a custom *model*, *host*, *endpoint* or all of them?

**Comment by t18n** [comment]
For Open WebUI, I think starting with making it a PWA-compatible is the least effort for the majority of the reward.

**Reply by isatopus** [reply]
It´s a really great path. On Google Play Store there´s many popular "Lite" apps eg. uber, tiktok, booking. It´s all PWA.

**Reply by Anaphylaxis** [reply]
The PWA works fine for me and I see a manifest.json

**Comment by tjbck** [comment]
https://github.com/open-webui/open-webui/issues/8414

**Comment by abdulazizalmass** [comment]
honestly, having the app on android and ios will increase the usability of the application.  It won't be easy but for open source and privacy, it is worth it. Thank you. it is really amazing and I will wait

**Comment by AlbertoSinigaglia** [comment]
news on this? now on EU you can have IOS apps outside of the AppStore

**Comment by TheHunterDog** [comment]
I was just thinking about building a app for iOS and android that would connect to the openwebui api. Should i opensource the project fully just like openwebui? And what would you suggest for type of framework, my intuition is going to react native because of the low learning curve. 

If you have any thought or ideas let me know

**Reply by Xyz00777** [reply]
i dont have any experience with it, but swift could also be interesting based on the supported enddevices for the software developement

**Reply by t18n** [reply]
If you go the React Native route with less experience on the technology, I would highly recommend Expo. Otherwise Tauri 2 or Capacitor are really good options too :)

**Reply by TheHunterDog** [reply]
> i dont have any experience with it, but swift could also be interesting based on the supported enddevices for the software developement

I have experience with swift but think a cross platform solution such as flutter or react native would be best for all users.

**Reply by TheHunterDog** [reply]
> If you go the React Native route with less experience on the technology, I would highly recommend Expo. Otherwise Tauri 2 or Capacitor are really good options too :)

React native without a framework is not really recommended by the react team themselves, I am currently experimenting with a simple react native application but can also see limitations brought by expo..

**Reply by t18n** [reply]
What limitations are we talking about?

**Reply by XenoUniv3rse** [reply]
When developing this, would it be possible to add support for CF headers?

**Reply by TheHunterDog** [reply]
I will begin to develop the application, if you are looking forward to contribute or keep track of the development check it out here: https://github.com/Enumbit/openChatApp

**Reply by TheHunterDog** [reply]
> When developing this, would it be possible to add support for CF headers?

What headers did you have in mind?

**Reply by XenoUniv3rse** [reply]
> > When developing this, would it be possible to add support for CF headers?
> 
> 
> 
> What headers did you have in mind?

Just [HTTP headers](https://developers.cloudflare.com/fundamentals/reference/http-headers/) to authenticate zero trust without the need to login.

**Reply by imrane** [reply]
Build native if possible. Better performance and AI can do more heavy lifting.

**Reply by XenoUniv3rse** [reply]
@TheHunterDog any update on this? 

I see the repo is still empty. Just checking in 😁

**Reply by TheHunterDog** [reply]
> @TheHunterDog any update on this? 
> 
> I see the repo is still empty. Just checking in 😁

Hi! Sorry about the progress my life has been changing which is leading to less time. I reached out to openwebui for a collaboration but sadly no reply. In July the app will be a more priority. I have build a simple sample app for Apple intelligence models which I will expand to openwebui. The app will be called "chathur" suggestions will be welcome for a different name or features!

**Reply by TheHunterDog** [reply]
> Build native if possible. Better performance and AI can do more heavy lifting. 

It will be developed in swift and SwiftUI! Fully native, most items will have the basic looks at first, if you have any suggestions about design after the first mvp feel free to create an issue with a design screenshot.

**Comment by Maticcm** [comment]
I made a android app: https://github.com/Maticcm/Open-WebUI-Client-for-Android

**Comment by JamesClarke7283** [comment]
> **Is your feature request related to a problem? Please describe.** No
> 
> **Describe the solution you'd like** I know that I can just add the webpage to my home screen on iOS and will act as "Native" but I would like to have an actual application that I can download from the app store. After download, allow users to enter the address to their Open WebUI service. I think that will give users more of a "native" feel.
> 
> **Additional context** Thanks for all the hard work!! What an amazing project!

What you want(IMO) is it to be a full PWA(Progressive Web App).

OpenWebUI is a webapp (as the name suggests), making it also a PWA wouldn't be too much of a jump (compared to a native port, unless you are doing a cheap and dirty embed with reskin).

That way you can install it as a local web APP, you dont get the native look nessesarily, but all other boxes are ticked.
- Security (compartmented)
- Local/Offline (Workers)
- Truly Platform Indiependent
- Far less dev-time than needing to worry about OS-specific API's (can focus on the application more, stability).

This is my opinion of how a modern mobile applications should be, of course Apple have a rocky relationship on PWA's for some reason, so your mileage may vary with iOS.

And BTW a PWA shouldnt be any less feature/integration issue than a native app, with the new W3C standards.
Chromium based browsers should have first-class support already.

Relevent page:
https://caniuse.com/mdn-api_permissions

**Reply by XenoUniv3rse** [reply]
Yes and no. 

I have my OpenWebUI behind Azure SSO login. Having to keep signing into that first with authentication gets pretty annoying. Having a mobile app that can support HTTP headers will solve this issue.

**Reply by noahkiss** [reply]
Doesn't it already act like a PWA? Or what is the difference? On iOS, I can add it to my homescreen, and it launches in its own window, not in Safari. There are really only a couple issues with it - sometimes the keyboard / page scrolls off-screen, when doing voice-to-text the screen will lock at the most inopportune time, and on iPadOS 26 there's no way to open the sidebar on the left because the traffic light blocks touch interaction (although that may be on Apple to adjust). Otherwise it works (relatively) fine. 

I still 100% vote for a dedicated app, though - but can understand the reluctance to re-create the whole UI in Swift.

**Comment by ValerioNeriGit** [comment]
I am thinking of building a flutter app for OpenWebUI, It would be tremendously useful for me. I will make it opensource of course, and whoever want to help is welcome.

Do you guys know where I could find the complete api reference for auth, chats, session, models, etc..?

**Reply by TheHunterDog** [reply]
Hi @ValerioNeriGit, currently I am also working on a mobile app in native iOS if you would like I can share multiple endpoints I already have implemented. The chat function is a bit weird. Would love to connect!

**Reply by haydonryan** [reply]
Since openwebui supports the OpenAI api. Can we use the libraries built to support it? https://platform.openai.com/docs/libraries/python-library

**Reply by ValerioNeriGit** [reply]
@TheHunterDog absolutely, were can I contact you? 

@haydonryan thank you for sharing! I will take a look. I suppose for the chat itself probably we could use that one, but I still gonna need specifics for the other functionalities that are internal, like session, workspaces, chat branching etc. I did a couple of tests and I have a base with auth, list available chats and sending new messages on existing chat.

**Reply by TheHunterDog** [reply]
> @TheHunterDog absolutely, were can I contact you? 

Let's start with mail: mark@enumbit.com

**Reply by tockudex** [reply]
what about a matrix chatroom or so? so that interested people can follow your progress and ideas? :)

**Reply by rgaricano** [reply]
> what about a matrix chatroom or so? so that interested people can follow your progress and ideas? :)

you can use channels feature for that.

**Reply by ValerioNeriGit** [reply]
Hello everyone,
I’ve put together a first barebone version of the app: it already has login, model selection, new chat, and messages (branching and other advanced features are not there yet). https://github.com/ValerioNeriGit/open-webui-flutter 

There’s still a lot to build. I’ll keep working on it in my spare time, but I’d really appreciate your help: testing, suggesting what features should or shouldn’t be included, and contributing code. Once we reach a stable point, we can publish it on the store. Discussions are enabled on my repo so feel free to open one

Feel free to contribute, test, and share any feedback or improvements!

**Reply by XenoUniv3rse** [reply]
Thank you !!

**Comment by Felitendo** [comment]
I've found a mobile client for OpenWebUI that works great: https://github.com/cogwheel0/conduit

**Reply by haydonryan** [reply]
Looks interesting even with the $3.99 cost.  But why does the app list "voice calling"?

**Reply by Felitendo** [reply]
Because it has that feature (lol). It‘s speech to text (you talk to the AI) and then text to speech (the AI talks back to you).

<img width="1180" height="820" alt="IMG_0965" src="https://github.com/user-attachments/assets/bcd23f4f-e359-4aee-bf81-5ab9e9316167" />

**Reply by haydonryan** [reply]
They should have used different words to describe it imo. eg  Voice session

See I hear "calling" or "call" and think phone call.

**Reply by Zeeshan-system** [reply]
I found another IOS client for OWUI:
[https://github.com/Ichigo3766/Open-Relay](https://github.com/Ichigo3766/Open-Relay)

You can try this

**Reply by haydonryan** [reply]
I now use telegram directly to openclaw / custom app.

**Comment by veliseev93** [comment]
Open MobileUI — free mobile client for Open WebUI 🎉

We built Open MobileUI, a native iOS & Android app for your self-hosted Open WebUI. It's free and open source (GPL v3) — just install it, enter your server URL, sign in, and start chatting on mobile.

📦 Source: https://github.com/RonasIT/open-webui-react-native
📲 App Store: https://apps.apple.com/us/app/open-mobileui/id6754266176
📲 Google Play: https://play.google.com/store/apps/details?id=com.open.web.ui

**Reply by XenoUniv3rse** [reply]
> Open MobileUI — free mobile client for Open WebUI 🎉
> 
> 
> 
> We built Open MobileUI, a native iOS & Android app for your self-hosted Open WebUI. It's free and open source (GPL v3) — just install it, enter your server URL, sign in, and start chatting on mobile.
> 
> 
> 
> 📦 Source: https://github.com/RonasIT/open-webui-react-native
> 
> 📲 App Store: https://apps.apple.com/us/app/open-mobileui/id6754266176
> 
> 📲 Google Play: https://play.google.com/store/apps/details?id=com.open.web.ui

This is amazing, thank you!!! 

Does it support SSO login?

**Reply by Classic298** [reply]
<img width="877" height="273" alt="image" src="https://github.com/user-attachments/assets/f36abe81-6dc2-4096-beed-556777f7146b" />

**Reply by veliseev93** [reply]
> > Open MobileUI — free mobile client for Open WebUI 🎉
> > We built Open MobileUI, a native iOS & Android app for your self-hosted Open WebUI. It's free and open source (GPL v3) — just install it, enter your server URL, sign in, and start chatting on mobile.
> > 📦 Source: https://github.com/RonasIT/open-webui-react-native
> > 📲 App Store: https://apps.apple.com/us/app/open-mobileui/id6754266176
> > 📲 Google Play: https://play.google.com/store/apps/details?id=com.open.web.ui
> 
> This is amazing, thank you!!!
> 
> Does it support SSO login?

No, the app supports email/password login only

**Reply by XenoUniv3rse** [reply]
> > > Open MobileUI — free mobile client for Open WebUI 🎉
> 
> > > We built Open MobileUI, a native iOS & Android app for your self-hosted Open WebUI. It's free and open source (GPL v3) — just install it, enter your server URL, sign in, and start chatting on mobile.
> 
> > > 📦 Source: https://github.com/RonasIT/open-webui-react-native
> 
> > > 📲 App Store: https://apps.apple.com/us/app/open-mobileui/id6754266176
> 
> > > 📲 Google Play: https://play.google.com/store/apps/details?id=com.open.web.ui
> 
> > 
> 
> > This is amazing, thank you!!!
> 
> > 
> 
> > Does it support SSO login?
> 
> 
> 
> No, the app supports email/password login only

Are there plans to enable SSO login?

**Reply by veliseev93** [reply]
> > > > Open MobileUI — free mobile client for Open WebUI 🎉
> > 
> > 
> > > > We built Open MobileUI, a native iOS & Android app for your self-hosted Open WebUI. It's free and open source (GPL v3) — just install it, enter your server URL, sign in, and start chatting on mobile.
> > 
> > 
> > > > 📦 Source: https://github.com/RonasIT/open-webui-react-native
> > 
> > 
> > > > 📲 App Store: https://apps.apple.com/us/app/open-mobileui/id6754266176
> > 
> > 
> > > > 📲 Google Play: https://play.google.com/store/apps/details?id=com.open.web.ui
> > 
> > 
> > > 
> > 
> > 
> > > This is amazing, thank you!!!
> > 
> > 
> > > 
> > 
> > 
> > > Does it support SSO login?
> > 
> > 
> > No, the app supports email/password login only
> 
> Are there plans to enable SSO login?

Unfortunately, the Open WebUI API does not allow this to be implemented for the mobile app.

**Reply by XenoUniv3rse** [reply]
> > > > > Open MobileUI — free mobile client for Open WebUI 🎉

Conduit supports SSO login flow

**Reply by veliseev93** [reply]
> > > > > > Open MobileUI — free mobile client for Open WebUI 🎉
> 
> Conduit supports SSO login flow

Google SSO login is now available in v1.6.5 - https://github.com/RonasIT/open-webui-react-native/releases/tag/v1.6.5

**Comment by JoaoZaokk** [comment]
I was also missing a decent native Open WebUI experience on iOS, so I ended up building one.

It’s a native SwiftUI client, not a WebView, with streaming chat, attachments, image generation, RAG, voice mode, themes, and support for self-hosted Open WebUI servers.

You can build it from the repo, or there’s also a ready-to-use App Store version:

GitHub: https://github.com/JoaoZaokk/openwebui-ios
App Store: https://apps.apple.com/us/app/openwebui-mobile/id6783872393

**Comment by theoden8** [comment]
I've built a webview client that supports multiple instances (MIT).

Source: https://github.com/theoden8/webspace_app
App store: https://apps.apple.com/app/webspace-app/id6758049523
Google play: https://play.google.com/store/apps/details?id=org.codeberg.theoden8.webspace
F-Droid: https://f-droid.org/packages/org.codeberg.theoden8.webspace
